docs/developers_notes/basis_module.md

```diff
@@ -0,0 +1,130 @@
+2. Checks that the output of the method does not exceed `gb_limit` GB of memory size. This size limit is specified at class initialization.
+3. Calls the subclass-specific `_get_samples` method, which returns equidistant samples over the domain of the basis function. The domain may depend on the type of basis.
+4. Calls the subclass-specific `_evaluate` method.
+5. Returns both the sample grid points and the evaluation output at each grid point.
+
+### Abstract Methods
+
+The `neurostatslib.basis.Basis` class has the following abstract methods, which every non-abstract subclass must implement:
+
+1. `_evaluate`: Evaluates a basis over some specified samples.
+2. `_get_samples`: Returns a tuple of equidistant samples (as flat numpy arrays) for each dimension of the basis function. These samples are used to form a grid over which the basis is evaluated.
+3. `_check_n_basis_min`: Checks the minimum number of basis functions required. This requirement can be specific to the type of basis.
+
+
+## The `AdditiveBasis` Class
+
+This class extends the abstract `Basis` class and represents a set of basis functions that are combined additively. It overrides the `_evaluate`, `_get_samples`, and `_check_n_basis_min` methods with specific implementations for additive basis functions.
+
+### Overridden Methods
+
+1. `_evaluate`: For an `AdditiveBasis` object, this method should evaluate each basis function at the specified samples and then add the results together.
+2. `_get_samples`: This method should return equidistant samples over the domain of each basis function. The samples should then be summed together.
+3. `_check_n_basis_min`: This method checks whether the number of basis functions meets the minimum requirement for an additive basis set.
+
+## The `MultiplicativeBasis` Class
+
+This class extends the `Basis` abstract class and represents a set of basis functions that are combined multiplicatively. It overrides the `_evaluate`, `_get_samples`, and `_check_n_basis_min` methods with specific implementations for multiplicative basis functions.
+
+### Overridden Methods
+
+1. `_evaluate`: For a `MultiplicativeBasis` object, this method should evaluate each basis function at the specified samples and then multiply the results together.
+2. `_get_samples`: This method should return equidistant samples over the domain of each basis function. The samples should then be multiplied together.
+3. `_check_n_basis_min`: This method checks whether the number of basis functions meets the minimum requirement for a multiplicative basis set.
+
+## The `SplineBasis` and `MSplineBasis` Classes
+
+The `SplineBasis` class is an abstract subclass of `Basis` that represents a spline basis function. The `MSplineBasis` class is a concrete subclass of `SplineBasis` that implements a specific type of spline basis, the M-spline.
```

Comment:
needs a longer description or a link to what a "spline basis function" means. and what's shared among spline bases that makes it worth having their own subclass.